### ISSUE TITLE
Problems with tuning

### ISSUE THREAD

**Issue by MrGitGo** [Unity-Technologies/ml-agents#1805]
I have a shooter game where 3 agents try to shoot the player. In the training phase the player has random movements but in the normal game you have to control the player by yourself. The player can also shoot. This is how the game looks: 
![Part4](https://user-images.githubusercontent.com/33628476/54068405-b00b7780-424c-11e9-978a-61c96ee78cf9.PNG)

I try to config the reward and the yaml file to get the best results for my agents but after 8 million steps I wanted to post my progress here and want to ask how I can get better results. So I will explain in detail for what I am giving reward and how my yaml file looks like.

The game concept is just the three agents have to shoot the player and the player has to shoot them. 

yaml file:

```
default:
    trainer: ppo
    batch_size: 80
    beta: 3.0e-3
    buffer_size: 15240
    epsilon: 0.2
    gamma: 0.86
    hidden_units: 75
    lambd: 0.935
    learning_rate: 9.0e-3
    max_steps: 9.0e6
    memory_size: 256
    normalize: true
    num_epoch: 5
    num_layers: 1
    time_horizon: 100
    sequence_length: 64
    summary_freq: 1000
    use_recurrent: false
    use_curiosity: false
    curiosity_strength: 0.01
    curiosity_enc_size: 128
```
Academy:

![Part5](https://user-images.githubusercontent.com/33628476/54068515-d1209800-424d-11e9-895c-3736e03b14d4.PNG)

Brain:

![Part6](https://user-images.githubusercontent.com/33628476/54068517-dc73c380-424d-11e9-8d76-c2b713226925.PNG)

Observation:

![Part7](https://user-images.githubusercontent.com/33628476/54068524-f3b2b100-424d-11e9-8626-406933ec2b3b.PNG)

Rewards:

- As you can see in the game pic I have rays attached to the agents. So they get a reward if they look at the player: 
![Part8](https://user-images.githubusercontent.com/33628476/54068543-22c92280-424e-11e9-999e-9d246adfecb4.PNG)

- And I have a range reward. 
![Part9](https://user-images.githubusercontent.com/33628476/54068555-4ab88600-424e-11e9-9d0e-015e24438298.PNG)
If I didn't use this, the agents would't move around anymore. So I this this was neccessary to move them around.

- Penalty given each step to encourage agent to finish task quickly.
```
        AddReward(-0.03f / 100);
```

- give reward if hit player bit bullet. If not add negative reward:
 ```
if (checkIfhitPlayer())
                    {
                        AddReward(1f);
                    }
                    else
                        AddReward(-0.00001f);
```

- I have notices this now. I have one more method where I give the agents the same reward for shooting the player:

```
    public void AIHitThePlayer()
    {
        AddReward(1f);
    }
```

-    
```
public void PlayerHitTheAI()
    {
        AddReward(-1f);

        //Debug.Log("Player hit the Ai");

        gameObject.SetActive(false);

        //Done();        
    }
```

- If the time is up and the agents survived. They get reward(1f)

I think this is it with the rewards

This is is the tensorboard:

![Part1](https://user-images.githubusercontent.com/33628476/54068672-a33c5300-424f-11e9-8892-a5360703ed33.PNG)
![Part2](https://user-images.githubusercontent.com/33628476/54068673-a6cfda00-424f-11e9-819c-8ad505f2ecef.PNG)

And one more thing to say. I have trained the model with 12 agents not 3. At the very beginning I had 3 and the normal game will have 3 but I tried this model training with 12 agents and one player:
![Part3](https://user-images.githubusercontent.com/33628476/54068745-2c538a00-4250-11e9-84c6-245a29f7d6d4.PNG)

I hope you guys can help me. I am desperate. Thanks in advance!

**Comment by Semih67** [comment]
Hey @MarcoMeter  and @ervteng, could you help me with this? I am really desperate and I do not know what else I can change to improve the learning rate

**Comment by xiaomaogy** [comment]
Hi @MrGitGo I would suggest you start with the default hyperparameter instead of using your own hyperparameter, and first start with easier task to make it train, then go from there.

**Comment by xiaomaogy** [comment]
Thank you for the discussion. We are closing this issue due to inactivity. Feel free to reopen it if you’d like to continue the discussion though.
